### ISSUE TITLE
Way to build on ARM Linux

### ISSUE THREAD

**Issue by drencorxeen** [MyTDT-Mysoft/COCO-FastLoader#1]
Just asking if there is a way to get this project to build on a Raspberry Pi or other SBC ARM based system?

**Comment by MyTDT-Mysoft** [comment]
well i believe its possible, altough officially i dont know the state of freebasic for ARM, but i have some unnoficial build that i used to build for android, and it uses my own version of the freebasic runtime library (simplified over the monstruous official one), so probabily combining both, and giving the simplicity of the program (regarding rtlib requirements) i could provide a PI build... and later the toolchain to compile for that, but if the same sound library used on linux would work on PI or whatever ARM, then for sure my program can

**Comment by drencorxeen** [comment]
So I have been doing some digging and found that someone has been building FreeBasic on ARM as well.
https://www.freebasic.net/forum/viewtopic.php?f=18&t=26315&p=258867&hilit=raspberry#p258867
So now the next thing I am going to have to try is your code on my Raspberry Pi 3B+.
I am not sure if the audio out on the Raspberry Pi 3B+ can do the speed that is needed or not, but I do have a CreativeLabs SoundBlaster USB audio device I could try using as well.

I would like to see if it would be possible to use both DriveWire 4 and your package to make the Raspberry Pi 3B+ be usable for both options. :D

**Comment by MyTDT-Mysoft** [comment]
i see, let me know if it works fine, i think the internal audio of PI uses PWM which may or not work... but for sure it should work at normal speed (44khz) for double speed (88khz) you probabily would also need to recompile the library used for audio... as i think the library on linux have "fixed output frequency" (default 44khz)

**Comment by MyTDT-Mysoft** [comment]
ok as for review on this... latest freebasic version have a proper package for raspbian
https://sourceforge.net/projects/fbc/files/Binaries%20-%20Linux/raspbian9/

so just need to see if the support for one of the audio backend is good enough on pi to work with your card i assume its the default card so it should work, keep me updated if you try to compile it there... i plan to try on my pi3 as well altough i only have the built-in card

**Comment by drencorxeen** [comment]
Yeah I still need to download FreeBASIC for ARM and play with it on there. :)  So many projects trying to tinker with and also depends on mood too.

Right now in the mood for cassette port experiments. :)

**Comment by MyTDT-Mysoft** [comment]
> Yeah I still need to download FreeBASIC for ARM and play with it on there. :) So many projects trying to tinker with and also depends on mood too.
> 
> Right now in the mood for cassette port experiments. :)

have you tried it on Pi? now i have a Pi-Zero and a Pi-3 so i could even try myself, but my CP-400 is in another state... so i would need to feed to the emulator i guess hehe
